File: C-Sharp/Universal.cs
using System.Diagnostics;

namespace C_Sharp
{
    class UniversalTelemetry
    {
        private static Process p = Process.GetCurrentProcess();
        public Stopwatch tick = new Stopwatch();
        public PerformanceCounter cpuCounter = new PerformanceCounter("Processor", "% Processor Time", "_Total");
        public PerformanceCounter ramCounter = new PerformanceCounter("Process", "Working Set", p.ProcessName);
    }
}

File: C++CLI/Universal.cpp
#include "stdafx.h"

using namespace System::Diagnostics;

UniversalTelemetry::UniversalTelemetry()
{
	p = Process::GetCurrentProcess();
	tick = gcnew Stopwatch();
	cpuCounter = gcnew PerformanceCounter("Processor", "% Processor Time", "_Total");
	ramCounter = gcnew PerformanceCounter("Process", "Working Set", p->ProcessName);
}